packages/firebase_auth/android/src/main/java/io/flutter/plugins/firebaseauth/FirebaseAuthPlugin.java

```diff
@@ -491,8 +491,8 @@ private void handleReauthenticateWithCredential(
     AuthCredential credential = getCredential((Map<String, Object>) call.arguments());
 
     currentUser
-        .reauthenticate(credential)
-        .addOnCompleteListener(new TaskVoidCompleteListener(result));
+        .reauthenticateAndRetrieveData(credential)
```

Comment:
How about to change the method name in the dart/android/ios side to keep the real representation of what is being called? instead of `reauthenticateWithCredential` => `reauthenticateAndRetrieveData` ?

Reply:
Maybe it could be used the name already used in the ios side: `reauthenticateAndRetrieveDataWithCredential `

Reply:
Whatever you think is the best.

Reply:
By looking better at the code I think that your proposal is better than mine. You are right, it's better to adhere to what is called from the Firebase Auth native library.

Reply:
Yes, it would be a breaking change.
Since the code is not working today, I wouldn't mind this breaking change... but I think @collinjackson or @kroikie are better suited to decide what is pertinent for this PR.

Reply:
In newer versions of the Firebase iOS Auth SDK,`reauthenticateWithCredential` and `reauthenticateAndRetrieveDataWithCredential ` both do the same thing and return a `FIRAuthDataResultCallback`. I think we should go with the non-deprecated `reauthenticateWithCredential` which is what this PR is already doing, and no name change is needed.

Reference: 
 https://firebase.google.com/docs/reference/ios/firebaseauth/api/reference/Classes/FIRUser#-reauthenticatewithcredential:completion:

Reply:
@collinjackson can you help me in this decision? Should I left all the names as they are or should I change them to make them more coherent with the real method called, making it a breaking change?